amsclient/amsclient.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package amsclient
+
+import (
+	"fmt"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	"github.com/rs/zerolog/log"
+
+	"github.com/RedHatInsights/insights-results-aggregator/types"
+)
+
+// AMSClient allow us to interact the AMS API
+type AMSClient struct {
+	connection *sdk.Connection
+}
+
+// DefaultClient is the AMSClient used by default when no other is specified
+var DefaultClient *AMSClient
+
+// NewAMSClient creates an AMSClient from the configuration
+func NewAMSClient(conf Configuration) (*AMSClient, error) {
+	conn, err := sdk.NewConnectionBuilder().
+		URL(conf.URL).
+		Tokens(conf.Token).
+		Build()
+
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to build the connection to AMS API")
+		return nil, err
+	}
+
+	return &AMSClient{
+		connection: conn,
+	}, nil
+}
+
+// func initDefaultClient() {
+// 	if DefaultClient != nil {
+// 		return
+// 	}
+
+// 	DefaultClient, err := NewAMSClient(conf.Configuration.AMSClientConf)
+// 	if err != nil {
+// 		log.Error().Err(err).Msg("Cannot init the default client for AMS API")
+// 	}
+// }
+
+// GetClustersForOrganization retrieves the clusters for a given organization using the default client
+func (c *AMSClient) GetClustersForOrganization(orgID types.OrgID) []types.ClusterName {
```

Comment:
Why don't you return an error too?

Reply:
ATM I don't see that necessary, probably returning an empty array will be enough. When I work on the integration of this library with the org_overview endpoint, probably it will be required, so it's very likely that you will be right :-)